hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlOperatorTable.java

```diff
@@ -402,6 +408,40 @@ private SqlCall rewriteCall(SqlCall call) {
             return call;
         }
 
+        @SuppressWarnings("CheckStyle")
+        private SqlCall expandView(SqlCall call) {
+            if (call instanceof SqlSelect) {
+                SqlSelect selectCall = (SqlSelect) call;
+                ViewResolver viewResolver = validator.getViewResolver();
+
+                if (selectCall.getFrom() == null) {
+                    return call;
+                }
+
+                SqlNode from = selectCall.getFrom();
+                if (from instanceof SqlIdentifier) {
```

Comment:
We need to handle the JOIN case

Reply:
Created an issue for follow-up PR: https://github.com/hazelcast/hazelcast/issues/19972